library/ssl_cli.c

```diff
@@ -2459,9 +2467,15 @@ static int ssl_rsa_encrypt_partial_pms( mbedtls_ssl_context *ssl,
         {
             ssl->handshake->premaster_generated = MBEDTLS_SSL_FI_FLAG_SET;
         }
+        else
+        {
+            ret = MBEDTLS_ERR_PLATFORM_FAULT_DETECTED;
+            goto fault_detected;
+        }
     }
     else
     {
+fault_detected:
```

Comment:
@piotr-now , I prefer not to jump to "else", lets put the debug print in place and jump to regular clean